hyperspy/ipython_profile/hyperspy_magics.py

```diff
@@ -0,0 +1,30 @@
+from IPython.core.magic import (Magics, magics_class, line_magic,
+                                cell_magic, line_cell_magic)
+
+
+@magics_class
+class HyperspyMagics(Magics):
+
+    @line_magic
+    def hyperspy(self, line):
+        sh = self.shell
+        first_import_part = ("import numpy as np\n"
+                             "import hyperspy.hspy as hs\n")
+        exec(first_import_part, sh.user_ns)
+
+        if len(line) == 0:
+            toolkit = hs.preferences.General.default_toolkit
+        else:
+            toolkit = line
+
+        sh.enable_matplotlib(toolkit)
+
+        second_import_part = "import matplotlib.pyplot as plt\n"
+        exec(second_import_part, sh.user_ns)
+
+        sh.set_next_input("# %hyperspy " + toolkit + "\n" + first_import_part +
+                          "%matplitlib " + toolkit + "\n" + second_import_part,
```

Comment:
Ouch, sorry for that - I forgot to push the commit. 

Let me know if the usage is annoying for you (as far as I know, I'm the only one using the magic + button combo). 

The button and magic work for the notebooks, however you have to be careful not to use magic with other code in the cell - it just replaces it. You can, of course, undo it (CTRL+z), but that's not very convenient.

The magic also works in consoles, but, as already mentioned in previous comments, leaves the "replaced" code in the not-run state... Which (at least for me) requires pressing return twice more. At the very least it's very verbose :)